File: Software Development Module/Revision/Encoder/EncoderApp.java
/*
EncoderApp.java
Billy Cussen
03/08/2020
*/

import java.util.Scanner;
import java.util.LinkedList;

public class EncoderApp{

    public static void main(String [] args){

        //Variables
        String inputWord = new String();
        String encodedWord;
        String repeatCheck = new String();
        boolean inputCheck;

        //Object
        Scanner keyboard;
        Encoder myEncoder;
        LinkedList<String> encodeHistory = new LinkedList<String>();

        do{
            repeatCheck = new String();
            keyboard = new Scanner(System.in);

            //Input
            do{
                inputCheck = true;

                System.out.print("Please enter the word to be encoded: ");
                inputWord = keyboard.nextLine();

                for(int i = 0; i < inputWord.length();i++){
                    if(inputWord.charAt(i)==' '){
                        System.out.println("You need to enter a valid word (No Spaces)");
                        inputCheck = false;
                        break;
                    }
                }

            }while(!inputCheck);

            //Process
            myEncoder = new Encoder();
            encodedWord = myEncoder.getEncodedWord(inputWord);
            encodeHistory.add("Input Word: "+inputWord+", Encoded Word: "+encodedWord);
            myEncoder = null;

            //Output
            System.out.println(encodedWord);

            do{
                inputCheck = true;
                System.out.print("Would you like to retry (Enter Y or N)");
                repeatCheck = keyboard.nextLine();

                if(!(repeatCheck.equalsIgnoreCase("Y")||repeatCheck.equalsIgnoreCase("N"))){
                    System.out.println("You need to enter Y or N!");
                    inputCheck = false;
                }

            }while(!inputCheck);

            keyboard = null;

        }while(repeatCheck.equalsIgnoreCase("Y"));

        System.out.println("\nHISTORY");
        for(String s : encodeHistory){
            System.out.println(s);
        }

        encodeHistory = null;
    }

}

File: Python Course/Software Development Module - in Python/Revision/Encoder/EncoderApp.py
"""
EncoderApp.py
Billy Cussen
22/10/2020
"""

#Objects
import Encoder
from Encoder import myEncoder
import gc

#Variables
repeat = "y"
wordHistory = []

while repeat == "y":

    #Input - Enter Word to be encoded
    inputCheck = False
    while inputCheck == False:
        try:
            word = str(input("Please enter the word you would like to encode: "))
            if len(word) <= 0 or " " in word:
                raise Exception
        except Exception as e:
            print("You need to enter a valid word (no spaces!)")
        else:
            inputCheck = True

    #Process & Output
    myEncoderApp = myEncoder(word)
    result = myEncoderApp.encodeWord()
    print(result)
    wordHistory.append(result)
    myEncoderApp = None
    gc.collect()
    
    #Input - Repeat?
    repeat = ""
    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter \"y\" to repeat and \"n\" to exit: "))
        repeat = repeat.lower()
        if repeat != "y" and repeat != "n":
            print ("Please enter \"y\" or \"n\"")

print("***WORD HISTORY***")
for w in wordHistory:
    print (w)

wordHistory = None
gc.collect()
print("\nThanks for using my Encoder App!")